Shopify/cli
# Opt-in to deployments prompt as part of the app deploy command

### WHY are these changes introduced?

Fixes https://github.com/Shopify/app-deploys/issues/516



### WHAT is this pull request doing?
- Move logic to handle deployment mode to it own module
- Updates the info and warning banner for legacy and unified deploy respectively
- Integrates the [new parters API](https://github.com/Shopify/partners/pull/48049) to enable to roll unified deployment up
- Add a new info message banner to the `select` and `autocomplete` prompts



### How to test your changes?
- (oprional) In case the [partners PR](https://github.com/Shopify/partners/pull/48049) is not merged, create one spin instance `spin up partners -c partners.branch=499-opt-in-mutation`
- Run the command `npm run shopify app deploy` (if you use spin `NODE_TLS_REJECT_UNAUTHORIZED=0 SHOPIFY_SERVICE_ENV=spin SPIN_INSTANCE=$(spin show -o name) npm run shopify app deploy`), you should see the new legacy banner and a prompt to roll unified up.

![image](https://github.com/Shopify/cli/assets/105213827/c9ec2cc2-2baf-4406-993e-22bbd3ebd485)

- Confirm the roll up and and the upgrading should be executed so you'll se the new banner as well

![image](https://github.com/Shopify/cli/assets/105213827/085d0fbb-cb75-4451-af6f-c3de11db2469)

- In case you want to revert to the legacy deployment, access the app betas section in partners internals and disable `app_unified_deployment`

![image](https://github.com/Shopify/cli/assets/105213827/4336e092-b157-4140-aa75-5fec9fbe958b)



### Post-release steps



### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
